### DISCUSSION TITLE
Home Assistant Versions and Foxess_Modbus Updates

### DISCUSSION THREAD

**Discussion by gjhill6100** [nathanmarlor/foxess_modbus#698] [General]
If I want to update from my current HA OS version 9.5 to 10.5 and HA Core 2023.3.6 to 2024.10.3, is there anything I need to know with my Foxess_Modbus installation before I do this ?

I am currently running FoxEss_Modbus version v1.7.6

What is the correct process to update the system and not crash everything ?? :)

Thanks for any tips / advice

**Comment by FozzieUK** [comment]
The OS upgrades are none breaking, but you must update to the latest version of the Foxess Modbus integration v1.12.2 before you start the upgrade to your HA Core as there have been a number of breaking changes since 2023.x

**Reply by FozzieUK** [reply]
Actually just checked the minimum HA core version for the latest integration which is 2023.7 - so you’ll need to do the Core update then after restarting immediately update the integration.

**Reply by gjhill6100** [reply]
Hi,
Many thanks for the update. So would the update sequence be:

1. Update HA Core to 2023.7
2. Reboot
3. Update HA OS Version to 10.5
4. Reboot
5. Update Foxess Modbus integration to v1.12.2
6. Reboot
7. Update HA Core to 2024.10.3
8. Reboot

??

**Reply by FozzieUK** [reply]
1. Update to HA OS 10.5 (you can actually do this at anytime, but get it out of the way before you start).
2. Reboot
3. Update straight to HA Core 2024.10.3
4. Reboot
When it restarts the Modbus integration will no longer work and will error at startup - ignore this
5. Immediately update to v1.12.2
6. Reboot

It will all now be working on the latest OS.

**Comment by gjhill6100** [comment]
Hi,

I’m doing the update, does all of the historical data get lost ?

Many thanks

**Reply by FozzieUK** [reply]
No you won’t lose history, but always make sure you take backups before you do any major work on HA.

**Comment by letsstartbeinganon** [comment]
How do I update the FoxEss version? My integration says it's running on 1.0.0?

**Reply by robc22** [reply]
I am seeing the same issue with the FoxEss version, I have recently updated to 2025.1 and this broke the integration and I think the latest version of the FoxEss resolves this but cannot figure out how to manually force the update.

Any help gratefully received.

Thanks

**Reply by letsstartbeinganon** [reply]
I definitely updated my FoxEss integration via the Home Assistant settings page, which notified me FoxEss had an update available. All appears to be working although when you go the FoxEss integration page, it does say 1.0.0 there.

**Reply by canton7** [reply]
> although when you go the FoxEss integration page, it does say 1.0.0 there.

Can you share a screenshot?

**Reply by letsstartbeinganon** [reply]
Of course, see attached.
![Screenshot_1](https://github.com/user-attachments/assets/e3953b41-2a30-429d-8ed2-39dd561fe3d6)

**Comment by allseven** [comment]
I have the same problem with both FoxESS_Modbus and FoxESS_Em...

I've installed Core 2025.1.4 but FoxESS_Modbus and FoxESS_Energy Management refuse to update to the latest versions. I have deleted them, re-downloaded them, reinstalled them and restarted the installation multiple times to no avail.

Is it possible to force them to update?

Current installation is:
Core 2025.1.4
Supervisor 2024.12.3
Operating System 14.2
Frontend 20250109.2
FoxESS_em 0.0.2
FoxESS_modbus 1.0.0

![Firefox_Screenshot_2025-02-02T18-27-33 653Z](https://github.com/user-attachments/assets/6a23ecdc-ed20-4bd9-a7d0-273c88d824d5)
![Firefox_Screenshot_2025-02-02T18-28-00 455Z](https://github.com/user-attachments/assets/5e80b7f2-dc40-40b0-bc49-b1fa16c1e176)
![Firefox_Screenshot_2025-02-02T18-26-49 758Z](https://github.com/user-attachments/assets/04a4df64-ff3d-4df8-9a72-319e56d45f0a)

**Comment by allseven** [comment]
I have the same problem with both FoxESS_Modbus and FoxESS_Em...

I've installed Core 2025.1.4 but FoxESS_Modbus and FoxESS_Energy Management refuse to update to the latest versions. I have deleted them, re-downloaded them, reinstalled them and restarted the installation multiple times to no avail.

Is it possible to force them to update?

Current installation is:
Core 2025.1.4
Supervisor 2024.12.3
Operating System 14.2
Frontend 20250109.2
FoxESS_em 0.0.2
FoxESS_modbus 1.0.0

**Reply by canton7** [reply]
How exactly are you trying to update foxess_modbus, and what are you seeing which makes you think that hasn't worked?

**Reply by allseven** [reply]
Settings > Integrations > Devices & services > delete the integration using its entry > restart HA > settings > Devices & services > add integration.

![Firefox_Screenshot_2025-02-02T18-26-49 758Z](https://github.com/user-attachments/assets/9b0585c9-f154-4702-bea2-fc24f6598139)
![Firefox_Screenshot_2025-02-02T18-27-33 653Z](https://github.com/user-attachments/assets/52a61561-f075-4a67-9cfe-6ff9d069a2d2)
![Firefox_Screenshot_2025-02-02T18-28-00 455Z](https://github.com/user-attachments/assets/80fe0b0f-d1da-4ec6-b579-c801d4ff63e3)

**Reply by canton7** [reply]
Did you not install it using HACS, following the [instructions in the README](https://github.com/nathanmarlor/foxess_modbus?tab=readme-ov-file#installation)?

It sounds like you manually downloaded an old version to the custom_components folder in your HA config. That won't automatically update. We recommend using HACS to install third-party integrations: that will also provide updates.

**Reply by canton7** [reply]
Ah I take that back -- I didn't recognise the new "Downloaded" section form HACS.

It looks like the version being displayed on the integration page is just wrong. I'll try and find out where HACS is getting that from.

**Reply by canton7** [reply]
In other words, I think foxess_modbus has successfully updated just fine -- the problem is just a problem displaying the version number. The fact that it's even starting is a strong indication, since old versions don't start at all on HA 2025.1 and above.

**Reply by allseven** [reply]
I've restarted HA again and I can see some of the cards using the Modbus and EM integrations starting to populate.

However, some are stuck on 'Entity not available': 'Entity not available: sensor.battery_soc' being an example. If I go into the card, delete a few characters at the end of the entity's name and then start to reinstate them, HA offers the full entity name (with the blue background) and the entity becomes available despite there being no actual change in the its name. If you see what I mean!

**Reply by canton7** [reply]
I'm afraid I have no idea what's going on there!

**Reply by canton7** [reply]
Are those entities listed if you click on the "1 service" in your screenshot?

**Reply by FozzieUK** [reply]
I think some of the entity names have been duplicated and have been replaced as sensor.’xxxxx’_2 - when you go to the card and delete a few characters the friendly name will be shown (which will be the same as it was before) but the new entity_id will actually have a ‘_2’ appended to the end.

You will either have to go through all the entities you use and update the entity names manually (including those you use in EM config such as sensor.battery_soc or the easiest way is to fall back to a backup - repeat the upgrade but this time delete the foxess modbus integration first and restart - then upgrade your HA core to latest and when it starts install the latest version of the foxess modbus integration.

**Reply by canton7** [reply]
You shouldn't need to delete anything. Just upgrade in HACS and restart.

**Reply by allseven** [reply]
They are but were disabled. I've enabled them now and some work. Others don't. I've also found the Charge Period Card won't install despite having been downloaded.

I haven't a clue what's going on!

**Reply by allseven** [reply]
Despite the progress, both integrations are still displaying as 
FoxESS_em 0.0.2
FoxESS_modbus 1.0.0

**Reply by canton7** [reply]
When you say "won't install", what exactly do you see which makes you think that it's failing?

Again, don't worry about the version numbers being displayed. Those are incorrect.

**Reply by allseven** [reply]
It appears that it was installed but, after a lot of trial and error I have a working card.

It had repeatedly failed to find the inverter (retaining a previous reference from before the installation of the updated Modbus integration?) and the card type was missing from the card definition.  I could copy this from the show code option and then insert it into the failed visual option. If, again, you see what I mean!

![Firefox_Screenshot_2025-02-02T22-06-21 878Z](https://github.com/user-attachments/assets/b198b1b6-adef-4427-ae8e-c251d4f57283)

**Reply by canton7** [reply]
If ifs recorded the wrong inverter, you can just reconfigure it? I'm afraid I'm not sure what's going wrong with HA trying to show the card: the card hasn't changed in a long time, so any problems see presumably with HA itself.

**Reply by allseven** [reply]
It certainly does look like it's HA but at least this time I've tried with 2025.1.4 I've got a lot more of it working. Time for some time away from it! Many thanks for your help: appreciated!

**Comment by gjhill6100** [comment]
My Charge Period Card has got errors recently with the attached error
message

**Reply by canton7** [reply]
I think you forgot to attach the error message?

Note that you cannot reply to the notification email with attachments, you must use the web interface.

**Reply by gjhill6100** [reply]
I will try again .. :)
![IMG_4941](https://github.com/user-attachments/assets/2a8f928b-5b39-4f5b-ad8c-ed86e6914123)

**Reply by canton7** [reply]
Try clearing your browser cache? I've had that one happen before.

**Reply by gjhill6100** [reply]
> Try clearing your browser cache? I've had that one happen before.

It maybe that HACS is having a problem loading ??

Any ideas  ?
![HACS Screenshot 2025-02-03 151410](https://github.com/user-attachments/assets/79d1cded-4f0f-4a4e-a130-912916a1e4cb)

**Reply by canton7** [reply]
I'd certainly start there: there's a fair chance that's related! Check the logs as it says.
